packages/app/src/cli/models/app/app.ts

```diff
@@ -101,6 +103,11 @@ export function appIsLaunchable(app: AppInterface) {
   return Boolean(frontendConfig || backendConfig)
 }
 
+export function includeConfigOnDeploy(configuration: AppConfiguration) {
+  if (isLegacyAppSchema(configuration)) return false
+  return configuration.build?.include_config_on_deploy && useVersionedAppConfig()
+}
```

Comment:
don't forget to move this to the app class

Reply:
I'm going to refactor the access to the new new flag `include_config_on_deploy`  when the `configuration` extensions are loaded. 
The next [PR](https://github.com/Shopify/cli/pull/3262) introduces the remote app beta flag `versioned_app_config` which will gate the new features as well. To avoid some hard conflicts I will address that refactor in that PR